components/remote-environment-broker/internal/mapping/mapping_service_test.go

```diff
@@ -0,0 +1,65 @@
+package mapping
+
+import (
+	"testing"
+	"time"
+
+	"github.com/kyma-project/kyma/components/remote-environment-broker/pkg/apis/applicationconnector/v1alpha1"
+	fake2 "github.com/kyma-project/kyma/components/remote-environment-broker/pkg/client/clientset/versioned/fake"
+	"github.com/kyma-project/kyma/components/remote-environment-broker/pkg/client/informers/externalversions"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func TestMappingService_ListEnvironmentMappings(t *testing.T) {
+	// GIVEN
+	em1 := fixEM("em1", "prod")
+	em2 := fixEM("em2", "prod")
+	emQa := fixEM("em3", "qa")
+	client := fake2.NewSimpleClientset(em1, em2, emQa)
+	informerFactory := externalversions.NewSharedInformerFactory(client, 0)
+	emInformer := informerFactory.Applicationconnector().V1alpha1().EnvironmentMappings().Informer()
+
+	waitForInformerStart(t, emInformer)
+	svc := newMappingService(emInformer)
+
+	// WHEN
+	result, err := svc.ListEnvironmentMappings("prod")
+
+	// THEN
+	require.NoError(t, err)
+	assert.Equal(t, result[0], em1)
+	assert.Equal(t, result[1], em2)
+	assert.Len(t, result, 2)
```

Comment:
1. first, check len of result, then access via indexes. 
2. But it will be better to not make assumptions about the order of returned objects, so use Contains method instead.